addons/ondevice-controls/src/types/Date.tsx

```diff
@@ -23,82 +23,66 @@ const Label = styled.Text(({ theme }: any) => ({
   color: theme.labelColor || 'black',
 }));
 
-// TODO seconds support
-class DateType extends PureComponent<
-  DateProps,
-  { isTimeVisible: boolean; isDateVisible: boolean }
-> {
-  static defaultProps = {
-    knob: {},
-    onChange: (value) => value,
-  };
-
-  static serialize = (value) => String(value);
-
-  static deserialize = (value) => parseFloat(value);
-
-  constructor(props) {
-    super(props);
-    this.state = {
-      isDateVisible: false,
-      isTimeVisible: false,
-    };
-  }
+const DateType = ({ onChange, arg: { name, value } }: DateProps) => {
+  const [isDateVisible, setIsDateVisible] = useState(false);
+  const [isTimeVisible, setIsTimeVisible] = useState(false);
 
-  showDatePicker = () => {
-    this.setState({ isDateVisible: true });
+  const showDatePicker = () => {
+    setIsDateVisible(true);
   };
 
-  showTimePicker = () => {
-    this.setState({ isTimeVisible: true });
+  const showTimePicker = () => {
+    setIsTimeVisible(true);
   };
 
-  hidePicker = () => {
-    this.setState({ isDateVisible: false, isTimeVisible: false });
+  const hidePicker = () => {
+    setIsDateVisible(false);
+    setIsTimeVisible(false);
   };
 
-  onDatePicked = (date) => {
-    const value = date.valueOf();
-    const { onChange } = this.props;
-    onChange(value);
-    this.hidePicker();
+  const onDatePicked = (pickedDate) => {
```

Comment:
Perhaps we could type this 🤔 

```suggestion
  const onDatePicked = (pickedDate: Date) => {
```